pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -528,6 +550,30 @@ func (a FeatureGateAnonymizer) Marshal(_ context.Context) ([]byte, error) {
 	return runtime.Encode(serializer, a.FeatureGate)
 }
 
+// IngressAnonymizer implements serialization with marshalling
+type ImageRegistryAnonymizer struct {
+	*registryv1.Config
+}
+
+// Marshal implements serialization of Ingres.Spec.Domain with anonymization
+func (a ImageRegistryAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	a.Spec.HTTPSecret = anonymizeString(a.Spec.HTTPSecret)
+	if a.Spec.Storage.S3 != nil {
+		a.Spec.Storage.S3.Bucket = anonymizeString(a.Spec.Storage.S3.Bucket)
+		a.Spec.Storage.S3.KeyID = anonymizeString(a.Spec.Storage.S3.KeyID)
```

Comment:
I don't know what's the purpose of anonymizeString, should it be used for anything that may identify the customer? RegionEndpoint may contain address of an in-house S3-compatible storage.

Reply:
yes, anynymize will hide the data (replace with `x`)